### ISSUE TITLE
Current latest container throws `Illegal instruction (core dumped)` & boot loops

### ISSUE THREAD

**Issue by CrystalSpore** [PhasecoreX/docker-red-discordbot#67]
**Describe the bug**

Bot starts up after updating to new container, pulls new dependencies for venv, then after setting prefix there is an `Illegal instruction (core dumped)`, at which point, the docker container fails & restarts

**Run command**
```
The Docker command/docker-compose.yml you are using to start the bot.
Be sure to censor your bot token (or any other sensitive info) if it is present!
```

NOTE: This docker container is managed by portainer. I got the equivalent run command using: `docker run --rm -v /var/run/docker.sock:/var/run/docker.sock nexdrew/rekcod lgbteacafe-redbot`

Command:
`docker run --name lgbteacafe-redbot --privileged --runtime runc -v /mnt/lgbteacafe_redbot:/data --net bridge --restart unless-stopped --security-opt 'label=disable' -h 56fc80b2f219 -l maintainer='PhasecoreX' -l org.opencontainers.image.created='2022-08-10T21:45:24.901Z' -l org.opencontainers.image.description='The newest Red-Discordbot in a convenient container' -l org.opencontainers.image.licenses='GPL-3.0' -l org.opencontainers.image.revision='852716026e48274eae57620235146201b36ffc89' -l org.opencontainers.image.source='https://github.com/PhasecoreX/docker-red-discordbot' -l org.opencontainers.image.title='Red-DiscordBot' -l org.opencontainers.image.url='https://github.com/PhasecoreX/docker-red-discordbot' -l org.opencontainers.image.vendor='PhasecoreX' -l org.opencontainers.image.version='master' -e 'GPG_KEY=E3FF2839C048B25C084DEBE9B26995E310250568' -e 'LANG=C.UTF-8' -e 'PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' -e 'PCX_DISCORDBOT_BUILD=2835915560' -e 'PCX_DISCORDBOT_COMMIT=852716026e48274eae57620235146201b36ffc89' -e 'PCX_DISCORDBOT_TAG=audio' -e 'PGID=1001' -e 'PREFIX=~' -e 'PUID=1000' -e 'PYTHON_GET_PIP_SHA256=d0b5909f3ab32dae9d115aa68a4b763529823ad5589c56af15cf816fca2773d6' -e 'PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/aeca83c7ba7f9cdfd681103c4dcbf0214f6d742e/public/get-pip.py' -e 'PYTHON_PIP_VERSION=22.0.4' -e 'PYTHON_SETUPTOOLS_VERSION=58.1.0' -e 'PYTHON_VERSION=3.9.13' -e 'SODIUM_INSTALL=system' -d --entrypoint "user-entrypoint" docker.io/phasecorex/red-discordbot:latest '/app/start-redbot.sh'`

**Environment info:**
What system are you running Docker on?

Ubuntu 22.04.2 LTS VM, running within Proxmox. Portainer used as a frontend for managing my Docker containers

**Additional context**
Add any other context about the problem here. A more in-depth description, logs, screenshots, whatever.

3 sets of logs:

First time seeing this issue:
[_lgbteacafe-redbot_logs.txt](https://github.com/PhasecoreX/docker-red-discordbot/files/11401958/_lgbteacafe-redbot_logs.txt)

After restarting the Ubuntu Host (to verify that it wasn't due to a failed network mount for persistent disk, etc):
[_lgbteacafe-redbot_logs_pt2.txt](https://github.com/PhasecoreX/docker-red-discordbot/files/11401961/_lgbteacafe-redbot_logs_pt2.txt)

After deleting venv folder, per instructions for issuing a bug report:
[_lgbteacafe-redbot_logs_pt3.txt](https://github.com/PhasecoreX/docker-red-discordbot/files/11401962/_lgbteacafe-redbot_logs_pt3.txt)

**Comment by Lifeismana** [comment]
What is the arch (amd64/arm64...) of your host ?

**Comment by PhasecoreX** [comment]
Yeah we need that information, and it's also weird that Portainer is generating the PYTHON_VERSION of 3.9.13, as that isn't the version used in the latest tag (should be 3.11.x).

**Comment by bolex17** [comment]
Getting the same error running on Ubuntu 20.04 running via docker-compose.

**Comment by PhasecoreX** [comment]
Can you try using the `core-audio-py39` tag? That one has an i386 architecture on it that I am not able to yet build with the newer python 3.11 ones. Wondering if your processor somehow needs that to work correctly. Redbot 3.5 can still work with python 3.9.

**Comment by CrystalSpore** [comment]
Currently trying that out.

It looks like the Env Variables are being loaded from this UI section, which I am assuming are "sticky", & don't automatically update when the docker container updates (arguably a failing on Portainer's Part)
![image](https://user-images.githubusercontent.com/6373584/236506066-0fd7cbe5-ee07-4884-a1d1-e761d4a08e0f.png)

**Comment by PhasecoreX** [comment]
Huh, that's weird. My next two things to test are:

1. See if setting environment variable `REDBOT_VERSION: ==3.4.19` and also pointing your /data mount elsewhere (to not overwrite the redbot 3.5 one) and using the -py39 tags works correctly.

2. See if you can just run Redbot 3.5 outside of the container, like a normal install. Just see if it starts as a fresh install, no need to move any personal data over to it.

Basically we want to know if your processor for some reason doesn't support a dependency used in redbot 3.5 (or discord.py 2).

**Comment by CrystalSpore** [comment]
Here is the run command (different bot called sporebot instead of lgbteacafe, but still using redbot):

`docker run --name sporebot-redbot --runtime runc -v /tmp/sporebot_redbot:/data --net bridge --restart unless-stopped -h 686f1c5c4830 -l maintainer='PhasecoreX' -l org.opencontainers.image.created='2022-08-04T21:45:16.316Z' -l org.opencontainers.image.description='The newest Red-Discordbot in a convenient container' -l org.opencontainers.image.licenses='GPL-3.0' -l org.opencontainers.image.revision='852716026e48274eae57620235146201b36ffc89' -l org.opencontainers.image.source='https://github.com/PhasecoreX/docker-red-discordbot' -l org.opencontainers.image.title='Red-DiscordBot' -l org.opencontainers.image.url='https://github.com/PhasecoreX/docker-red-discordbot' -l org.opencontainers.image.vendor='PhasecoreX' -l org.opencontainers.image.version='master' -e 'GPG_KEY=E3FF2839C048B25C084DEBE9B26995E310250568' -e 'LANG=C.UTF-8' -e 'PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' -e 'PCX_DISCORDBOT_BUILD=2799813587' -e 'PCX_DISCORDBOT_COMMIT=852716026e48274eae57620235146201b36ffc89' -e 'PCX_DISCORDBOT_TAG=audio' -e 'PGID=1001' -e 'PREFIX=!' -e 'PUID=1000' -e 'PYTHON_GET_PIP_SHA256=d0b5909f3ab32dae9d115aa68a4b763529823ad5589c56af15cf816fca2773d6' -e 'PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/aeca83c7ba7f9cdfd681103c4dcbf0214f6d742e/public/get-pip.py' -e 'PYTHON_PIP_VERSION=22.0.4' -e 'PYTHON_SETUPTOOLS_VERSION=58.1.0' -e 'PYTHON_VERSION=3.9.13' -e 'SODIUM_INSTALL=system' -e 'REDBOT_VERSION===3.4.19' -a stdout -a stderr --entrypoint "/bin/user-entrypoint" docker.io/phasecorex/red-discordbot:core-audio-py39 '/app/start-redbot.sh'`

In the UI I added the REDBOT_VERSION like this, which does seem to be pulling the older version of redbot:
![image](https://user-images.githubusercontent.com/6373584/236538094-d10981db-fc6e-49a0-9ba1-369e1bc2f401.png)

Failed on First Run because I forgot to set the bot token (since it isn't on disk anymore), but otherwise it seemed to pull everything fine:
[_sporebot-redbot_logs_no_token.txt](https://github.com/PhasecoreX/docker-red-discordbot/files/11409192/_sporebot-redbot_logs_no_token.txt)

Next run did work:
[_sporebot-redbot_logs_2nd_run.txt](https://github.com/PhasecoreX/docker-red-discordbot/files/11409240/_sporebot-redbot_logs_2nd_run.txt)

Potential next troubleshooting steps: Should I switch back to Latest image, & try to upgrade the version of Python in the ENV Variables? Essentially, should I get rid of all env variables except for the ones that I have explicitly assigned (such as prefix, PUID/PGID) & let it pull fresh from the container that it pulls?

**Comment by CrystalSpore** [comment]
I went ahead & did that, & it seems that all fresh variables were pulled down. So this is something I'll just need to keep in mind with Portainer in the future.

![ENV_Variables_Fresh_Pull](https://user-images.githubusercontent.com/6373584/236543951-dd49f1c4-a869-4ecc-9b67-e0f17e0c587e.png)

However, unfortunately, this resulted in an `Illegal instruction (core dumped)` once again
[_sporebot-redbot_logs_latest.txt](https://github.com/PhasecoreX/docker-red-discordbot/files/11409327/_sporebot-redbot_logs_latest.txt)

**Comment by Lifeismana** [comment]
You should try to run it outside of docker to male sure, it's not a problem woth docker

**Comment by CrystalSpore** [comment]
Yup, seems that it is something with Redbot on my system. I'll head over there for troubleshooting.

Sorry for wasting everyone's time here.

/tmp$ redbot sporebot
Red - Discord Bot | Configuration process

Please enter a valid token.
You can find out how to obtain a token with this guide:
https://docs.discord.red/en/stable/bot_application_guide.html#creating-a-bot-account
> <TOKEN_HERE>
Pick a prefix. A prefix is what you type before a command. Example:
!help
^ The exclamation mark is the prefix in this case.
The prefix can be multiple characters. You will be able to change it later and add more of them.
Choose your prefix:

Prefix> !
[20:57:19] INFO     [red] Begin updating guild configs to support multiple mod/admin roles
           INFO     [red] Done updating guild configs to support multiple mod/admin roles
           INFO     [red] Moving shared API tokens to a custom group
           INFO     [red] Migrating help menus to enum values
           INFO     [discord.client] logging in using static token
Illegal instruction (core dumped)

**Comment by CrystalSpore** [comment]
Hopped into the redbot discord, & did some troubleshooting. It seems that something with the `orjson` python package is crashing on my machine, but redbot should be falling back to a different json package for normal operations & it is not. If orjson gets uninstalled, then the bot starts up (but haven't done thorough testing with what else would be affected by having orjson removed). The redbot devs are going to do some more looking into this.

Should this ticket be closed out then?

**Comment by PhasecoreX** [comment]
Yeah I'll close it out since it isn't specific to this Docker image. Glad it's being looked at by the Redbot devs though! Seems like a simple try and except fix, depending on how integrated this new orjson thing is.
